src/ApiOauth2FlowDocument.js

```diff
@@ -0,0 +1,226 @@
+    const authorizationUri = this._computeAuthorizationUri(flow);
+    const authorizationGrant = this._computeAuthorizationGrant(flow);
+    let scopes = this._computeScopes(flow);
+    if (scopes && !(scopes instanceof Array)) {
+      scopes = [scopes];
+    }
+    this.accessTokenUri = accessTokenUri;
+    this.authorizationUri = authorizationUri;
+    this.authorizationGrant = authorizationGrant;
+    this.scopes = scopes;
+  }
+
+  /**
+  * Computes value for `accessTokenUri` property.
+  * @param {Object} flow OAuth2 flow from AMF model.
+  * @return {String|undefined}
+  */
+  _computeAccessTokenUri(flow) {
+    const flows = this._getValueArray(flow, this.ns.aml.vocabularies.security.flows);
+    if (flows) {
+      flow = flows[0]
+    }
+    return this._getValue(flow, this.ns.aml.vocabularies.security.accessTokenUri);
+  }
+  /**
+   * Computes value for `authorizationUri` property.
+   * @param {Object} flow OAuth2 flow from AMF model.
+   * @return {String|undefined}
+   */
+  _computeAuthorizationUri(flow) {
+    const flows = this._getValueArray(flow, this.ns.aml.vocabularies.security.flows);
+    if (flows) {
+      flow = flows[0]
+    }
+    return this._getValue(flow, this.ns.aml.vocabularies.security.authorizationUri);
+  }
+  /**
+   * Computes value for `authorizationGrant` property.
+   * @param {Object} flow OAuth2 flow from AMF model.
+   * @return {String|undefined}
+   */
+  _computeAuthorizationGrant(flow) {
+    return this._getValue(flow, this.ns.aml.vocabularies.security.flow);
+  }
+  /**
+   * Computes value for `scopes` property.
+   * @param {Object} flow OAuth2 flow from AMF model.
+   * @return {Array<Object>|undefined}
+   */
+  _computeScopes(flow) {
+    const flows = this._getValueArray(flow, this.ns.aml.vocabularies.security.flows);
+    if (flows) {
+      flow = flows[0]
+    }
+    if (!this._hasType(flow, this.ns.aml.vocabularies.security.OAuth2Flow) && !this._hasType(flow, this.ns.aml.vocabularies.security.OAuth2Settings)) {
+      return;
+    }
+    const key = this._getAmfKey(this.ns.aml.vocabularies.security.scope);
+    const scopes = this._ensureArray(flow[key]);
+    if (!scopes) {
+      return;
+    }
+    return scopes.map((item) => {
+      return {
+        description: this._computeDescription(item),
+        label: this._getValue(item, this.ns.aml.vocabularies.core.name)
+      };
+    });
+  }
+
+  /**
+   * Maps grant value to a user-friendly label
+   * @param {String} grant 
+   * @return {String}
+   */
+  _getLabelForGrant(grant) {
+    switch (grant) {
+      case 'implicit':
+        return 'Implicit';
+      case 'authorization_code':
+        return 'Authorization code';
+      case 'password':
+        return 'Password';
+      case 'client_credentials':
+        return 'Client credentials';
+      default:
+        return grant;
+    }
+  }
+
+  render() {
+    return html`<style>${this.styles}</style>
+    <div>
```

Comment:
I don't see point of putting it into a block element. This is not React :) The element itself is a block element (via `:host` style) and it doesn't need to render another element in the tree.

Reply:
Old habits die hard I guess :)